### DISCUSSION TITLE
Allow "Retry" and "Quit" buttons on the pause menu act as hold buttons

### DISCUSSION THREAD

**Discussion by BunnyTub** [ppy/osu#30801] [Ideas]
First off, I play with the Tap-X play style, meaning that I click with the tip of my pen, same as if I used the keyboard.
However, the tip of my pen is mapped to the left mouse button, and can interact with the UI like a mouse can.

The problem I'm having here, is that when I go to pause a beatmap to quickly do something else, I sometimes accidentally still try to click hit circles or sliders, and end up restarting or even quitting the map, losing my progress.

Replacing these with hold buttons would require the user to hold them to retry or quit, similar to dangerous buttons in some UI dialogs, such as beatmap deletion dialogs, and would add delay before fully losing progress on a beatmap.

**Comment by eatcakeisstupid** [comment]
writing this part at the end, i want to say this is a good idea no matter what but these are just my thoughts as far as tapping with pen goes

i've done the same for almost 5 years for all 2d games/anything on my pc and i like this but not the "hold buttons" part, i dunno, i have my hold to confirm time at 0ms because the quick restart button isn't so quick... sorry, i don't know how to say i'd rather a confirmation screen instead of that idea without an awkward justification because i've wished there was a confirmation for it for a while
i mean... what if you held down by accident? sure you get set it to 500ms but quick restarting just feels terrible

i haven't hit restart or quit by accident EVER though, even if you click down on "restart", you can just continue holding down and move your cursor off of the buttons

what i do as a solution currently is find a long slider/small break to pause on or just pause at the same time i hit a circle but just like before, incase the circle is on the retry button and i accidentally hit pause first, i don't lift up my pen and drag it to the bottom of the screen

**Reply by eatcakeisstupid** [reply]
ooh i have better idea, you need to double click restart/quit
maybe it can be a toggle idk
